stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -366,44 +387,80 @@ public struct Unsafe${Mutable}BufferPointer<Element>
     return UnsafeBufferPointerIterator(_position: _position, _end: _end)
   }
 
-  /// A pointer to the first element of the buffer.
-  ///
-  /// If the `baseAddress` of this buffer is `nil`, the count is zero. However,
-  /// a buffer can have a `count` of zero even with a non-`nil` base address.
   @_inlineable
-  public var baseAddress: Unsafe${Mutable}Pointer<Element>? {
-    return _position
+  public
+  func deallocate()
+  {
+    self._position?.deallocate()
   }
 
-  /// The number of elements in the buffer.
-  ///
-  /// If the `baseAddress` of this buffer is `nil`, the count is zero. However,
-  /// a buffer can have a `count` of zero even with a non-`nil` base address.
+% if Mutable:
+
   @_inlineable
-  public var count: Int {
-    switch(_position, _end) {
-    case (.some(let pos), .some(let end)):
-      return (end - pos)
-    case _:
-      return 0
+  public static
+  func allocate(capacity:Int) -> UnsafeMutableBufferPointer<Element>
+  {
+    let size = MemoryLayout<Element>.stride * capacity
+    let raw  = Builtin.allocRaw(size._builtinWordValue, Builtin.alignof(Element.self))
+               Builtin.bindMemory(raw, capacity._builtinWordValue, Element.self)
+    return UnsafeMutableBufferPointer(start: UnsafeMutablePointer(raw),
+                                      count: capacity)
+  }
+
+  // this is more efficient than the generic Sequence-based implementation
+  // because it assigns in one loop instead of making n one-item loops.
+  /*
+  @_inlineable
+  public
+  func assign(from source:UnsafeBufferPointer<Element>)
+  {
+    _debugPrecondition(source.count >= self.count,
```

Comment:
Ditto here about stdlib style and parentheses; I'll not repeat comments above about the styling of `guard...else` statements, which also apply here.